### DISCUSSION TITLE
ChadrcConfig type hints with lua_ls and lazydev.nvim

### DISCUSSION THREAD

**Discussion by realhackcraft** [NvChad/NvChad#2915] [Q&A] [answered]
I want to get type hints when editing chadrc.lua. I have ---@type ChadrcConfig right before local M = {} and I added "ui", which is where the ChadrcConfig type is defined, to library table of lazydev. Still, there is a diagnostics warning saying "Undefined type or alias `ChadrcConfig`."

**Comment by realhackcraft** [comment]
<img width="671" alt="Screenshot 2024-07-05 at 11 15 58" src="https://github.com/NvChad/NvChad/assets/65378120/b0c4400b-e344-43c2-9387-696b345768d7">
<img width="643" alt="Screenshot 2024-07-05 at 11 16 15" src="https://github.com/NvChad/NvChad/assets/65378120/e7a293db-f329-48d5-8249-25c7de3bc2a0">

**Reply by siduck** [reply]
because you didnt import our types well :/ 

check the  our default lspconfig lua_ls types table, idk about lazydev.nvim so cant help with it

**Reply by realhackcraft** [reply, marked as answer]
I changed "ui" to "ui/nvchad_types" and it worked!
